### PR TITLE
feat: [IAC-348]: Add iacm stage icons

### PR DESCRIPTION
**Pull request by willd0yle**
Add IACM stage icons:
- [x] Resource stack tab icon
- [x] Provision icon
- [x] Destroy icon
You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- StyleLint: `retrigger stylelint`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- ImageSnapshot `retrigger ImageSnapshot`

### CODE REVIEW

File: packages/icons/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@harness/icons",
-  "version": "1.111.0",
+  "version": "1.111.1",
```
Comment (knagurski) [inline packages/icons/package.json:3]: If this is adding something and not _only_ fixing a bug, it should be a minor release, not a point release
```suggestion
  "version": "1.112.0",
```

File: packages/icons/src/resource-stack.svg [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<svg viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (knagurski) [inline packages/icons/src/resource-stack.svg:1]: Worth running these through an optimiser ((ImageOptim)[https://imageoptim.com/mac] is my favourite). A quick test shows a 1/3rd reduction in size.

File: packages/icons/package.json (hunk 1 above)
Comment (willd0yle) [reply packages/icons/package.json]: done

File: packages/icons/src/resource-stack.svg (hunk 2 above)
Comment (willd0yle) [reply packages/icons/src/resource-stack.svg:1]: will do that now

**Review by ishantmehtaharness — APPROVED** [review]

**Pull request merged**